### PR TITLE
[indexer] - CI integration test

### PR DESCRIPTION
**Pull request by patrickkuo**
## Description 

Added skeleton CI integration test with postgesql DB, the test will wipe and rebuild the database using migration schemas and run test using the database.

## Test Plan 

CI

### CODE REVIEW

**Review by 666lcz — APPROVED** [review]
OMG this is amazing!!

File: crates/sui-indexer/Cargo.toml [hunk 1]
```diff
@@ -44,9 +44,13 @@ workspace-hack = { version = "0.1", path = "../workspace-hack" }
 
 fastcrypto = { workspace = true, features = ["copy_key"] }
 
+[features]
+pg_integration = []
+
 [dev-dependencies]
 test-utils = { path = "../test-utils" }
 sui-framework-build = { path = "../sui-framework-build" }
+diesel_migrations = "2.0.0"
```
[the remark this answers is not in the document]
Comment (patrickkuo) [reply crates/sui-indexer/Cargo.toml:53]: nice! thanks for the tips

[the remark this answers is not in the document]
Comment (patrickkuo) [reply crates/sui-indexer/Cargo.toml]: oh looks like this only works with [dependencies], getting `dev-dependencies are not allowed to be optional: `diesel_migrations``

[the remark this answers is not in the document]
Comment (after-ephemera) [reply crates/sui-indexer/Cargo.toml:53]: Ah lame okay. You might could use the `target.'cfg(feature = "pg_integration")'.dev-dependencies` thing described [here](https://stackoverflow.com/questions/61090286/cargo-toml-how-do-i-select-a-dependencys-feature-based-on-my-crates-features) but if not no problem.

**Pull request merged**